### PR TITLE
Fixing crashing bots / restarting bots

### PR DESCRIPTION
**Pull request by woblit**
Bug fixes to correct constantly resetting bots / crashing bots.

### CODE REVIEW

**Comment by loomsen** [comment]
Please target the beta branch for your PR.

**Comment by woblit** [comment]
Bug fixes should never really be targeted to a Beta branch redevelopment. Particularly when these fixes have been published and tested by other contributors prior to PR as per the contributor's private group. These should be targeted to Main as they are fixes to what currently breaks PCB functionality. New features etc should always be put into Beta, which is what I will be doing with the upcoming features that I have been putting together (with the help of some others).

**Comment by loomsen** [comment]
Well, hotfixes should target the main branch, I agree. But this is not a hotfix, but a bugfix. It's not like the bot doesn't work at all, is it? Basically you're improving the overall functionality. But I don't care, if @whittlem is fine with it, merge it to main.

File: models/Trading.py [hunk 1]
```diff
@@ -1455,22 +1455,30 @@ def __calculateSupportResistenceLevels(self):
...
+        try:
+            c1 = df["low"][i] < df["low"][i - 1]
+            c2 = df["low"][i] < df["low"][i + 1]
+            c3 = df["low"][i + 1] < df["low"][i + 2]
+            c4 = df["low"][i - 1] < df["low"][i - 2]
+            support = c1 and c2 and c3 and c4
+            return support
+        except:
+            support = False
+            return support
```
Comment (lazydroid) [inline models/Trading.py:1467]: one `return` ought to be enough, just pull it to the higher indentation level.

File: models/Trading.py [hunk 2]
```diff
@@ -1455,22 +1455,30 @@ def __calculateSupportResistenceLevels(self):
...
+        try:  
+            c1 = df["high"][i] > df["high"][i - 1]
+            c2 = df["high"][i] > df["high"][i + 1]
+            c3 = df["high"][i + 1] > df["high"][i + 2]
+            c4 = df["high"][i - 1] > df["high"][i - 2]
+            resistance = c1 and c2 and c3 and c4
+            return resistance
+        except:
+            resistance = False
+            return resistance
```
Comment (lazydroid) [inline models/Trading.py:1481]: same here with multiple returns

File: models/exchange/kucoin/api.py [hunk 3]
```diff
@@ -888,11 +888,16 @@ def getHistoricalData(
...
-            df["low"] = pd.to_numeric(df["low"])
-            df["high"] = pd.to_numeric(df["high"])
-            df["open"] = pd.to_numeric(df["open"])
-            df["close"] = pd.to_numeric(df["close"])
-            df["volume"] = pd.to_numeric(df["volume"])
+            #df["low"] = pd.to_numeric(df["low"])
+            #df["high"] = pd.to_numeric(df["high"])
+            #df["open"] = pd.to_numeric(df["open"])
+            #df["close"] = pd.to_numeric(df["close"])
+            #df["volume"] = pd.to_numeric(df["volume"])
```
Comment (lazydroid) [inline models/exchange/kucoin/api.py:895]: there's not much reason to keep dead code in the comments here. I'd rather remove the old code.

**Comment by woblit** [comment]
Looking into a few things - will create another PR once done - closing for now.

**Pull request closed without merging**